Resources/config/exporter.xml

```diff
@@ -1,6 +1,30 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <container xmlns="http://symfony.com/schema/dic/services" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
     <services>
-        <service id="sonata.core.exporter" class="Sonata\CoreBundle\Exporter\Exporter"/>
+        <service id="sonata.core.exporter.writer.csv" class="Exporter\Writer\CsvWriter" public="false">
+            <argument>php://output</argument>
+            <argument>,</argument>
+            <argument>"</argument>
+            <argument></argument>
+            <argument>true</argument>
+            <argument>true</argument>
+        </service>
+        <service id="sonata.core.exporter.writer.json" class="Exporter\Writer\JsonWriter" public="false">
+            <argument>php://output</argument>
+        </service>
+        <service id="sonata.core.exporter.writer.xls" class="Exporter\Writer\XlsWriter" public="false">
+            <argument>php://output</argument>
+        </service>
+        <service id="sonata.core.exporter.writer.xml" class="Exporter\Writer\XmlWriter" public="false">
+            <argument>php://output</argument>
+        </service>
+        <service id="sonata.core.exporter" class="Sonata\CoreBundle\Exporter\Exporter">
+            <argument type="collection">
+                <argument type="service" id="sonata.core.exporter.writer.csv" />
```

Comment:
Same here. You pass an array of service. Good.

But if I want to add one writer on it without dropping the default ones?

This is maybe for this @rande talked about factories...